config/robots.rb

```diff
@@ -2024,4 +2024,103 @@
                    target_state: 'passed'
                  }
                })
+
+  # Robots for pWGS 384 pipeline
+
+  custom_robot('star-pwgs-384-post-shear-to-pwgs-384-post-shear-xp',
+               name: 'STAR pWGS-384 Post Shear => pWGS-384 Post Shear XP',
+               beds: {
+                 bed(12).barcode => {
+                   purpose: 'LB Post Shear',
+                   states: ['passed'],
+                   label: 'Bed 12'
+                 },
+                 bed(13).barcode => {
+                   purpose: 'LB Post Shear',
+                   states: ['passed'],
+                   label: 'Bed 13'
+                 },
+                 bed(14).barcode => {
+                   purpose: 'LB Post Shear',
+                   states: ['passed'],
+                   label: 'Bed 14'
+                 },
+                 bed(15).barcode => {
+                   purpose: 'LB Post Shear',
+                   states: ['passed'],
+                   label: 'Bed 15'
+                 },
+                 bed(7).barcode => {
+                   purpose: 'pWGS-384 Post Shear XP',
+                   states: ['pending'],
+                   label: 'Bed 7',
+                   parents: [bed(12).barcode, bed(13).barcode, bed(14).barcode, bed(15).barcode],
+                   target_state: 'passed'
+                 }
+               })
+
+  custom_robot('bravo-pwgs-384-post-shear-xp-to-pwgs-384-end-prep',
+               name: 'Bravo pWGS-384 Post Shear XP => pWGS-384 End Prep',
+               require_robot: true,
+               beds: {
+                 bed(4).barcode => {
+                   purpose: 'pWGS-384 Post Shear XP',
+                   states: ['passed'],
+                   label: 'Bed 4'
+                 },
+                 car('1,4').barcode => {
+                   purpose: 'pWGS-384 End Prep',
+                   states: ['pending'],
+                   label: 'Carousel 1,4',
+                   target_state: 'started'
+                 },
+                 car('3,5').barcode => {
+                   purpose: 'pWGS-384 AL Lib',
+                   states: ['pending'],
+                   label: 'Carousel 3,5',
+                   target_state: 'started'
+                 }
+               })
+
+  custom_robot('bravo-pwgs-384-end-prep',
+               name: 'Bravo pWGS-384 End Prep',
+               verify_robot: true,
```

Comment:
I'm trying to understand how is going to verify the robot. It should be scanning its barcode, but how is it mapped with the name?

Reply:
As far as I can tell looking at other pipelines, specifically heron_384, and from how the UI behaves, the `require_robot` and `verify_robot` attributes are required when performing different steps on the same robot. This is something that we require as part of this pipeline.

From what I understand, including the `require_robot: true` attribute registers the first robot, with subsequent `verify_robot: true` attributes indicating that the steps are carried out on this same robot

Reply:
Yep, there's a box for scanning the robot, and I think if you put 'require_robot: true', the page makes sure you've filled that field in. 
Then, as you say, the verify_robot on a later step checks that plate has been previously verified on the same robot.